storage/store.go

```diff
@@ -308,7 +344,7 @@ func (sc *StoreContext) setDefaults() {
 }
 
 // NewStore returns a new instance of a store.
-func NewStore(ctx StoreContext, eng engine.Engine, nodeDesc *proto.NodeDescriptor) *Store {
+func NewStore(ctx StoreContext, eng engine.Engine, nodeDesc *proto.NodeDescriptor, stopper *util.Stopper) *Store {
```

Comment:
We shouldn't pass a stopper to both NewStore and Store.Start. Either save the stopper in NewStore and use it in Start, or delay construction of the scanner until Start (I prefer the latter).